src/generator/mod.rs

```diff
@@ -0,0 +1,73 @@
+use super::{Block, ListItem, Span};
+
+trait IteratorMyExt<I> where I : Iterator {
```

Comment:
This is pretty clever but maybe you should give it a better name like, uh, "Join"? "JoinExt"?